src/frontend/sharedComponents/BGMapCard.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from "react";
+import { defineMessages, useIntl } from "react-intl";
+import { StyleSheet, View, Text } from "react-native";
+import { TouchableOpacity } from "react-native-gesture-handler";
+import MapboxGL from "@react-native-mapbox-gl/maps";
+import { StackNavigationProp } from "react-navigation-stack/lib/typescript/src/vendor/types";
+
+import { LIGHT_GREY, MEDIUM_GREY } from "../lib/styles";
+import { ViewStyleProp } from "../sharedTypes";
+import { Pill } from "./Pill";
+import LocationContext from "../context/LocationContext";
+
+const m = defineMessages({
+  currentMap: {
+    id: "sharedComponents.BGMapCard.currentMap",
+    defaultMessage: "Current Map",
+  },
+  abbrevMegaBite: {
+    id: "sharedComponents.BGMapCard.abbrevMegaBite",
+    defaultMessage: "MB",
+    description: "The abbreviation for mega bite",
+  },
+});
+
+//ToDo: API calls to get styleURL, zoom level, center coordinate, etc.
+
+interface BGMapCardProps {
+  mapId: string;
+  mapTitle: string;
+  mapSize: number;
+  navigation: StackNavigationProp;
+  style?: ViewStyleProp;
+  onPress?: (() => void) | null;
+}
+
+export const BGMapCard = ({
+  mapSize,
+  mapTitle,
+  style,
+  onPress,
+  mapId,
+  navigation,
+}: BGMapCardProps) => {
+  const { formatMessage: t } = useIntl();
+  const { navigate } = navigation;
+  const { position } = React.useContext(LocationContext);
+  const [styleUrl, setStyleUrl] = React.useState<string>(
+    "mapbox://styles/mapbox/streets-v11"
+  );
+  const [zoomLevel, setZoomLevel] = React.useState<number>(6);
+  const [centerCoordinate, setCenterCoordinate] = React.useState<
+    (number | undefined)[]
+  >([position?.coords.longitude, position?.coords.latitude]);
+
+  function onPressDefault() {
+    navigate("OfflineAreas", { mapId });
+  }
+
+  React.useEffect(() => {
+    function getStyleURL() {
+      //To do: API call to get styleURL
+      return "mapbox://styles/mapbox/streets-v11";
+    }
+    function getZoomLevel() {
+      //To do: API call to get zoom level
+      //This should be min zoom. Where is this coming from?
+      return 6;
+    }
+    function getCenterCoordinate() {
+      //To do: API call to get center coordinate
+      //I believe this should come from the styleJSON `center` property
+      return [position?.coords.longitude, position?.coords.latitude];
+    }
+
+    setStyleUrl(getStyleURL());
+    setZoomLevel(getZoomLevel());
+    setCenterCoordinate(getCenterCoordinate());
+  }, []);
+
+  return (
+    <TouchableOpacity onPress={onPress || onPressDefault}>
+      <View style={[styles.container, style]}>
+        <MapboxGL.MapView
+          // placeholder style URL
+          styleURL={styleUrl}
+          compassEnabled={false}
+          zoomEnabled={false}
+          logoEnabled={false}
+          scrollEnabled={false}
+          style={[styles.map]}
+        >
+          <MapboxGL.Camera
+            zoomLevel={zoomLevel}
+            centerCoordinate={centerCoordinate}
+          />
```

Comment:
think we want to disable the animation here, via `animationDuration={0}` and (maybe, but not as necessary?) `animationMode="linearTo"`. curious about the `allowUpdates` prop too as it might be something we'd want to set to `false` for this component